formatTest/typeCheckedTests/expected_output/mlSyntax.re

```diff
@@ -100,3 +100,9 @@ let module EM = {
 };
 
 exception Ealias = EM.E;
+
+let switch_ = "switch";
+
+let pub_ = "public";
```

Comment:
Wait, I see a bug. What if the original `.ml` file had an identifier named `pub_`, and another identifier named `pub`. They both will get converted to `pub_` - which could not only fail to compile, but could also cause an unintentional bug. We always need to make sure that:

1. Every AST can be parsed in each syntax.
2.  In each syntax, `Parser(Printer(AST)) == AST` for all valid `AST`, for all syntaxes.

I think in this case for an AST where an identifier is "pub_" *in the AST*, it will be printed as `pub_`, and then it will be parsed into the AST as the identifier `pub`, which is not what we began with so it violated rule 2.

One easy way to that ensure that these rules hold, is we can simply *swap* the text *printing* of two already valid identifiers. For example, we could swap `mod` with `%`.
- When parsing the `%` token, we would always store it as identifier `"mod"` in the AST, and when we print we know to print the identifier `"mod"` as `%`. 
- When parsing the `mod` token, we would always store it as identifier `"%"` in the AST, and when we print we know to print the identifier `"%"` in the AST as `mod`.
- Therefore every AST is representable (rule number one), and `Parse(Printing(AST)) == AST`.

This case is a little different though because `%` and `mod` were both valid identifiers in each language. In our case, `public` is not a valid identifier in ML and `pub` is not a valid identifier in `Reason`. In either case, since ML is the canonical syntax, an AST that has an identifier named `"public"` is not a valid AST, so we don't need to even consider ASTs of that form.

That means, that we have one more option at our disposal. We can swap a *valid* identifier, with an identifier that could *never* occur. In `Reason`, we could parse the identifier `"public"` (as in `let public = "foo"`), and then store it in the AST as `"pub"`, then always ensure that `Reason` prints out the AST identifier `"pub"` as `"public"`. That allows every *valid* ML AST to be printed in Reason, and parsed back in Reason, and every valid `Reason` parse, to be printed in `ML` without any loss of information or changes in the AST. (note: If you parsed an ML file that had `let pub = ` it would be stored as `"pub"` in the AST and printed out in `Reason` as `let public =`.)